extensions/controllers/sandboxwarmpool_controller.go

```diff
@@ -0,0 +1,244 @@
+	client.Client
+}
+
+//+kubebuilder:rbac:groups=extensions.agents.x-k8s.io,resources=sandboxwarmpools,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=extensions.agents.x-k8s.io,resources=sandboxwarmpools/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile implements the reconciliation loop for SandboxWarmPool
+func (r *SandboxWarmPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	// Fetch the SandboxWarmPool instance
+	warmPool := &extensionsv1alpha1.SandboxWarmPool{}
+	if err := r.Get(ctx, req.NamespacedName, warmPool); err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Info("SandboxWarmPool resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get SandboxWarmPool")
+		return ctrl.Result{}, err
+	}
+
+	// Handle deletion
+	if !warmPool.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("SandboxWarmPool is being deleted")
+		return ctrl.Result{}, nil
+	}
+
+	// Save old status for comparison
+	oldStatus := warmPool.Status.DeepCopy()
+
+	// Reconcile the pool (create or delete Pods as needed)
+	if err := r.reconcilePool(ctx, warmPool); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Update status if it has changed
+	if err := r.updateStatus(ctx, oldStatus, warmPool); err != nil {
+		log.Error(err, "Failed to update SandboxWarmPool status")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// reconcilePool ensures the correct number of pods exist in the pool
+func (r *SandboxWarmPoolReconciler) reconcilePool(ctx context.Context, warmPool *extensionsv1alpha1.SandboxWarmPool) error {
+	log := log.FromContext(ctx)
+
+	// Get the pool label value (pool-name-hash)
+	poolLabelValue := warmPool.Name
+
+	// List all pods with the pool label
+	podList := &corev1.PodList{}
+	labelSelector := labels.SelectorFromSet(labels.Set{
+		poolLabel: poolLabelValue,
+	})
+
+	if err := r.List(ctx, podList, &client.ListOptions{
+		LabelSelector: labelSelector,
+		Namespace:     warmPool.Namespace,
+	}); err != nil {
+		log.Error(err, "Failed to list pods")
+		return err
+	}
+
+	desiredReplicas := warmPool.Spec.Replicas
+	currentReplicas := int32(len(podList.Items))
+
+	log.Info("Pool status",
+		"desired", desiredReplicas,
+		"current", currentReplicas,
+		"poolLabel", poolLabelValue)
+
+	// Update status replicas
+	warmPool.Status.Replicas = currentReplicas
+
+	var allErrors error
+
+	// Create new pods if we need more
+	if currentReplicas < desiredReplicas {
+		podsToCreate := desiredReplicas - currentReplicas
+		log.Info("Creating new pods", "count", podsToCreate)
+
+		for i := int32(0); i < podsToCreate; i++ {
+			// Generate deterministic name based on current count + index
+			podName := fmt.Sprintf("%s-pod-%d", warmPool.Name, currentReplicas+i)
```

Comment:
this seems not right, say a case: there are pod 0, 1,  3. But the desired pod numbers are 4, it will try to create pod-3